JumpscaleCore/servers/gedis/handlers.py

```diff
@@ -300,21 +302,35 @@ def _handle_request(self, request, address, user_session):
         elif request.command.command == "auth":
             tid, seed, signature = request.arguments
             tid = int(tid)
-            try:
-                tclient = j.clients.threebot.client_get(threebot=tid)
-            except Exception as e:
-                logdict = j.core.myenv.exception_handle(e, die=False, stdout=True)
-                return (logdict, None)
-            try:
-                verification = tclient.verify_from_threebot(seed, signature)
-            except Exception as e:
-                logdict = j.core.myenv.exception_handle(e, die=False, stdout=True)
-                return (logdict, None)
-            # if we get here we know that the user has been authenticated properly
-            user_session.threebot_id = tclient.tid
-            user_session.threebot_name = tclient.name
-            j.shell()
 
+            current_threebot_id = int(j.tools.threebot.me.default.tid)
+            # If working on same machine no need to get a client to authenticate
+            # otherwise, we'll have infinite loop
+            if current_threebot_id != tid:
+                try:
+                    tclient = j.clients.threebot.client_get(threebot=tid)
+                except Exception as e:
+                    logdict = j.core.myenv.exception_handle(e, die=False, stdout=True)
+                    return (logdict, None)
+                try:
+                    verification = tclient.verify_from_threebot(seed, signature)
+                except Exception as e:
+                    logdict = j.core.myenv.exception_handle(e, die=False, stdout=True)
+                    return (logdict, None)
+
+                # if we get here we know that the user has been authenticated properly
+                user_session.threebot_id = tclient.tid
```

Comment:
Like I mentioned above, the `client_get` will crash when tid is 0. But even if we get past this point somehow (by changing the logic), here we set the id and name of the threebot. In the case I mentioned above, we wont have values for them yet, which is a legit scenario. However, what are the implications of setting a user_session which empty values?

Reply:
let user_session empty will cause actors that needs authorization to fail to execute because, we are in the way for implementing actors that needs to recognize who's executing them and based on the proper permissions/acl refuse or grant the request
letting session empty is not a good thing and inconsistent with the auth commmand we do through the client

Reply:
There are some methods exposes by some actors that have no restriction on the user. "public" methods. 
what is the plan to handle these now that "authentication" seems to be automatic both on client and server side ?

Reply:
> let user_session empty will cause actors that needs authorization to fail to execute because, we are in the way for implementing actors that needs to recognize who's executing them and based on the proper permissions/acl refuse or grant the request
> letting session empty is not a good thing and inconsistent with the auth commmand we do through the client

Yea, it's normal if the actor fails if there is no user session. It's the user's responsibility to make sure they're executing actors they have access to and executing them after having the `me` instance configured. And like @zaibon mentioned, some actors are going to be public ex: the registeration I mentioned above.